django_cassandra_engine/models.py

```diff
@@ -1,5 +1,450 @@
+        # We should call the contribute_to_class method only if it's bound
+        if not inspect.isclass(value) and hasattr(value, 'contribute_to_class'):
+            value.contribute_to_class(cls, name)
+        else:
+            setattr(cls, name, value)
+
+    @classmethod
+    def _add_django_meta_and_register_model(cls, klass, attrs, name):
+        # Create the class.
+        module = attrs.pop('__module__')
+        new_class = klass
+        attr_meta = attrs.pop('Meta', None)
+        abstract = getattr(attr_meta, 'abstract', False)
+        if not attr_meta:
+            meta = getattr(new_class, 'Meta', None)
+        else:
+            meta = attr_meta
+
+        if meta:
+            meta.managed = False
+
+        app_label = None
+
+        # Look for an application configuration to attach the model to.
+        app_config = apps.get_containing_app_config(module)
+
+        if getattr(meta, 'app_label', None) is None:
+            if app_config is None:
+                if not abstract:
+                    raise RuntimeError(
+                        "Model class %s.%s doesn't declare an explicit "
+                        "app_label and isn't in an application in "
+                        "INSTALLED_APPS." % (module, name)
+                    )
+
+            else:
+                app_label = app_config.label
+
+        new_class.add_to_class(
+            '_meta', DjangoCassandraOptions(meta, app_label, cls=new_class))
+        new_class.add_to_class('_default_manager', new_class.objects)
+        new_class.add_to_class('_base_manager', new_class.objects)
+        new_class._meta.apps.register_model(new_class._meta.app_label, new_class)
+        return new_class
+
+    @classmethod
+    def check(cls, **kwargs):
+        errors = []
+        return errors
+
+
+class DjangoCassandraQuerySet(query.ModelQuerySet):
+    name = 'objects'
+    use_in_migrations = False
+
+    def order_by(self, *colnames):
+        order_using_python = False
+
+        if len(colnames) == 0:
+            clone = copy.deepcopy(self)
+            clone._order = []
+            return clone
+
+        conditions = []
+        for colname in colnames:
+            try:
+                conditions.append('"{0}" {1}'.format(
+                    *self._get_ordering_condition(colname))
+                )
+            except query.QueryException as err:
+                if 'Can\'t order on' not in err.message:
+                    # if the exception isn't due to ordering, raise it up!
+                    raise err
+
+                msg = (
+                    '.order_by() with column "{}" failed! '
+                    'falling back to ordering in python. '
+                    'Exception was:\n{}'
+                ).format(colname, err.message)
+                warnings.warn(msg)
+                order_using_python = True
+
+        clone = copy.deepcopy(self)
+        clone._order.extend(conditions)
+
+        if order_using_python is True:
```

Comment:
TODO: refactor, test, add setting to explicitly trigger this behavior else raise.